apps/wallet/src/ui/app/components/transactions-card/index.tsx

```diff
@@ -46,22 +47,29 @@ export function TransactionCard({ txn, address }: TransactionCardProps) {
     // Get the balance changes for the transaction and the amount
     const balanceChanges = getBalanceChangeSummary(txn, recognizedPackagesList);
 
-    function getAmount(tx: IotaTransactionBlockResponse) {
-        if ((isTimelockedStaking || isTimelockedUnstaking) && tx.events) {
-            return getTransactionAmountForTimelocked(
-                tx.events,
+    const [balance, coinType] = (() => {
+        // Use any non-iota coin type if found, otherwise simply use IOTA
+        const nonIotaCoinType = balanceChanges?.[address]
+            .map((change) => change.coinType)
+            .find((coinType) => coinType !== IOTA_TYPE_ARG);
```

Comment:
```suggestion
        // Use any non-iota coin type if found, otherwise simply use IOTA
        const nonIotaCoinType = balanceChanges?.[address]
            ?.map((change) => change.coinType)
            .find((coinType) => coinType !== IOTA_TYPE_ARG);
```

balanceChanges?.[address] might be undefined as well